src/App.js

```diff
@@ -0,0 +1,201 @@
+import React, { Component } from "react";
+
+import TodoItemsList from "./components/TodoItemsList";
+import TodoInput from "./components/TodoInput";
+import TodoSearch from "./components/TodoSearch";
+import "./App.css";
+
+class App extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      name: "",
+      surname: "",
+      message: "",
+      isSubmitted: false,
+      key: 2,
+      todoItems: [
+        {
+          name: "Andrew",
+          surname: "anime",
+          message: "Hello World!",
+          date: '12.10.2018',
+          key: 1,
+          isCompleted: false,
+          display: true,
+          textMatchStart: "",
+          textMatch: false,
+          textMatchEnd: ""
+        },
+        {
+          name: "Yura",
+          surname: "notanime",
+          message: "Hello World!",
+          date: '12.10.2018',
+          key: 2,
+          isCompleted: true,
+          display: true,
+          textMatchStart: "",
+          textMatch: false,
+          textMatchEnd: ""
+        }
+      ]
+    };
+  }
+
+  handleChange = e => {
+    this.setState({ [e.target.name]: e.target.value })
+  };
+
+  handleAdd = e => {
+    let newKey = this.state.key;
```

Comment:
keeping a key as a separate part of the state is a little weird because you need to make to update it accordingly on each add/delete.
Some suggestions: 
- use key as a combined of couple values of the item e.g. name + surname + message
- or use some utility to generate unique key each time e.g. [uuid](https://www.npmjs.com/package/uuid)

Reply:
I choose uuid, works fine.